### PR TITLE
Bluetooth: ascs: Fix Source ASE link loss state transition

### PR DESCRIPTION
**Pull request by MariuszSkamra**
According to the ASCS_v1.0 the ASE in Streaming state shall transit to QoS Configured state when link loss happen.

Relates: ES-24215 (errata)
Fixes: BAP/USR/SCC/BV-168-C

### CODE REVIEW

File: subsys/bluetooth/audio/ascs.c [hunk 1]
```diff
@@ -879,23 +872,22 @@ static void ascs_ep_iso_disconnected(struct bt_bap_ep *ep, uint8_t reason)
...
-	} else if (ep->status.state == BT_BAP_EP_STATE_DISABLING) {
-		/* CIS has been unexpectedly disconnected */
-		ase->unexpected_iso_link_loss = true;
+		ase->state_pending = BT_BAP_EP_STATE_QOS_CONFIGURED;
 
-		ascs_ep_set_state(ep, BT_BAP_EP_STATE_QOS_CONFIGURED);
+		err = k_work_submit(&ase->state_transition_work);
+		if (err < 0) {
+			LOG_ERR("Failed to submit state transition work err %d", err);
+		}
```
Comment (Thalley) [inline subsys/bluetooth/audio/ascs.c:890]: So I assume you are doing it this way, instead of `ascs_ep_set_state(ep, BT_BAP_EP_STATE_QOS_CONFIGURED);` because of the checks we have in `ascs_ep_set_state`?

What about we still set `ase->unexpected_iso_link_loss = true;` and add that as a check in `ascs_ep_set_state` to allow for `STREAMING->QOS_CONFIGURED` or `DISABLING->QOS_CONFIGURED` if `unexpected_iso_link_loss` is true?

Comment (MariuszSkamra) [reply subsys/bluetooth/audio/ascs.c]: That's a good idea

**Review by Thalley — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]
